bson/marshalvalue_cases_test.go

```diff
@@ -0,0 +1,127 @@
+	codeWithScope := primitive.CodeWithScope{Code: "code", Scope: D{{"a", "b"}}}
+	idx, scopeCore := bsoncore.AppendDocumentStart(nil)
+	scopeCore = bsoncore.AppendStringElement(scopeCore, "a", "b")
+	scopeCore, _ = bsoncore.AppendDocumentEnd(scopeCore, idx)
+	decimal128 := primitive.NewDecimal128(5, 10)
+	structTest := marshalValueStruct{Foo: 10}
+	structCore, err := Marshal(structTest)
+	assert.Nil(t, err, "Marshal error: %v", err)
+
+	return []marshalValueTestCase{
+		{"double", 3.14, bsontype.Double, bsoncore.AppendDouble(nil, 3.14)},
+		{"string", "hello world", bsontype.String, bsoncore.AppendString(nil, "hello world")},
+		{"binary", primitive.Binary{1, []byte{1, 2}}, bsontype.Binary, bsoncore.AppendBinary(nil, 1, []byte{1, 2})},
+		{"undefined", primitive.Undefined{}, bsontype.Undefined, []byte{}},
+		{"object id", oid, bsontype.ObjectID, bsoncore.AppendObjectID(nil, oid)},
+		{"boolean", true, bsontype.Boolean, bsoncore.AppendBoolean(nil, true)},
+		{"datetime", primitive.DateTime(5), bsontype.DateTime, bsoncore.AppendDateTime(nil, 5)},
+		{"null", primitive.Null{}, bsontype.Null, []byte{}},
+		{"regex", regex, bsontype.Regex, bsoncore.AppendRegex(nil, regex.Pattern, regex.Options)},
+		{"dbpointer", dbPointer, bsontype.DBPointer, bsoncore.AppendDBPointer(nil, dbPointer.DB, dbPointer.Pointer)},
+		{"javascript", primitive.JavaScript("js"), bsontype.JavaScript, bsoncore.AppendJavaScript(nil, "js")},
+		{"symbol", primitive.Symbol("symbol"), bsontype.Symbol, bsoncore.AppendSymbol(nil, "symbol")},
+		{"code with scope", codeWithScope, bsontype.CodeWithScope, bsoncore.AppendCodeWithScope(nil, "code", scopeCore)},
+		{"int32", 5, bsontype.Int32, bsoncore.AppendInt32(nil, 5)},
+		{"int64", int64(5), bsontype.Int64, bsoncore.AppendInt64(nil, 5)},
+		{"timestamp", primitive.Timestamp{T: 1, I: 5}, bsontype.Timestamp, bsoncore.AppendTimestamp(nil, 1, 5)},
+		{"decimal128", decimal128, bsontype.Decimal128, bsoncore.AppendDecimal128(nil, decimal128)},
+		{"min key", primitive.MinKey{}, bsontype.MinKey, []byte{}},
+		{"max key", primitive.MaxKey{}, bsontype.MaxKey, []byte{}},
+		{"struct", structTest, bsontype.EmbeddedDocument, structCore},
+		{"D", D{{"foo", int32(10)}}, bsontype.EmbeddedDocument, structCore},
+		{"M", M{"foo": int32(10)}, bsontype.EmbeddedDocument, structCore},
+		{"ValueMarshaler", marshalValueMarshaler{Foo: 10}, bsontype.Int32, bsoncore.AppendInt32(nil, 10)},
+	}
+
+}
+
+func marshalValueTestCases(t *testing.T) []marshalValueTestCase {
+	t.Helper()
+
+	marshalValueTestCases := unmarshalValueTestCases(t)
+
+	interfaceTest := marshalValueInterfaceOuter{
+		Reader: marshalValueInterfaceInner{
+			Foo: 10,
+		},
+	}
+	interfaceCore, err := Marshal(interfaceTest)
+	assert.Nil(t, err, "Marshal error: %v", err)
+
+	marshalValueTestCases = append(
+		marshalValueTestCases,
+		marshalValueTestCase{"interface", interfaceTest, bsontype.EmbeddedDocument, interfaceCore},
```

Comment:
What is the purpose of adding this here instead of with the other "marshalValueTestCase" in "unmarshalValueTestCases" ?